src/altair/epoch_processing.rs

```diff
@@ -77,8 +138,27 @@ pub fn process_rewards_and_penalties<
         MAX_VALIDATORS_PER_COMMITTEE,
         SYNC_COMMITTEE_SIZE,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<()> {
+    // No rewards are applied at the end of `GENESIS_EPOCH` because rewards are for work done in the previous epoch
+    let current_epoch = get_current_epoch(state, context);
+    if current_epoch == GENESIS_EPOCH {
+        return Ok(());
+    }
+
+    let mut deltas = Vec::new();
+    for flag_index in 0..crate::altair::PARTICIPATION_FLAG_WEIGHTS.len() {
+        let flag_index_delta = get_flag_index_deltas(state, flag_index, context)?;
+        deltas.push(flag_index_delta);
+    }
+    let mut inactivity_penalty_deltas = vec![get_inactivity_penalty_deltas(state, context)?];
+    deltas.append(&mut inactivity_penalty_deltas);
```

Comment:
I feel like we can skip the intermediate `Vec` allocation here

something like `deltas.append(get_inactivity_penalty_deltas(..)?);`